src/Tokenizer/Analyzer/ClassAnalyzer.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer;
+
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\TypeAnalysis;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @internal
+ */
+final class ClassAnalyzer
```

Comment:
ok, so this class is raised for 2.12 in this PR and master in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4275/files#diff-292cb19f45776d6735715655afbd167fR22

please, create an separated, standalone PR with only `ClassAnalyzer` for 2.12 line and apply changes requested in #4275

it will make whole process for sure a bit slower, but we will review it atomically and have no issues of having different version on 2.12 and master because of 2 different reviews

Reply:
Hi @keradus ,

I created a new PR #4362 it only contains the ClassAnalyzer.

This PR will be on hold until that new PR is merged.

Reply:
I will resolve this conversation, the issue is fixed, both PR has been updated.